### ISSUE TITLE
SortedBy only apply to headers and actual data

### ISSUE THREAD

**Issue by ShijunK** [FasterXML/jackson-dataformat-csv#96]
Follow-up to #42 

It seems the implementation only sorts the columns in headers, not actual values. 

I have a test for 2.6.3

````
   public void testValueReorderByName() throws Exception
    {
        CsvMapper mapper = mapperForCsv();
        CsvSchema schema = mapper.schemaFor(Mixed.class);
        assertEquals(aposToQuotes("['a','b','c','d']"), schema.getColumnDesc());
        schema = schema.sortedBy("b", "c");
        assertEquals(aposToQuotes("['b','c','a','d']"), schema.getColumnDesc());

        Mixed value = new Mixed();
        value.a = 1;
        value.b = 2;
        value.c = 3;
        value.d = 4;

        schema = schema.withHeader();
        String outputString = mapper.writer(schema).writeValueAsString(Arrays.asList(value));
        assertThat(outputString, is(equalTo("b,c,a,d\n2,3,1,4\n")));

        _verifyLinks(schema);
    }
````

test output:
````
java.lang.AssertionError: 
Expected: is "b,c,a,d\n2,3,1,4\n"
     but: was "b,c,a,d\n1,2,3,4\n"
````

**Comment by jspo** [comment]
I've found the same issue. The reason is that the method sortedBy only sorts the Column objects in the schema list but doesn't change the column objects. The Column objects of the list contain the original values of the _index and the _next attributes.

**Comment by cowtowncoder** [comment]
Yes, I can reproduce this with given test. Will try to see how to fix.

**Comment by ShijunK** [comment]
@cowtowncoder  great, thanks

**Comment by cowtowncoder** [comment]
Fix should be in 2.7.0 which was released couple of days ago. `2.6.5` will be released within week or two as well.
